packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue

```diff
@@ -142,6 +181,69 @@ export default defineComponent({
   },
   setup(props) {
     const { $gettext } = useGettext()
+    const visibleItems = ref<BreadcrumbItem[]>([])
+    const hiddenItems = ref<BreadcrumbItem[]>([])
+    const displayItems = ref<BreadcrumbItem[]>(props.items.slice())
+
+    const getBreadcrumbElement = (id): HTMLElement => {
+      return document.querySelector(`[data-item-id="${id}"]`)
+    }
+
+    const calculateTotalBreadcrumbWidth = () => {
+      let totalBreadcrumbWidth = 100 // 100px margin to the right to avoid breadcrumb from getting too close to the controls
+      visibleItems.value.forEach((item, index) => {
+        const breadcrumbElement = getBreadcrumbElement(item.id)
+        const itemClientWidth = breadcrumbElement?.getBoundingClientRect()?.width || 0
+        totalBreadcrumbWidth += itemClientWidth
+      })
+      return totalBreadcrumbWidth
+    }
+
+    const reduceBreadcrumb = (offsetIndex) => {
+      const breadcrumbMaxWidth = props.maxWidth
+      if (!breadcrumbMaxWidth) {
+        return
+      }
+      document.getElementById(props.id)?.style.setProperty('--max-width', `${breadcrumbMaxWidth}px`)
+      const totalBreadcrumbWidth = calculateTotalBreadcrumbWidth()
+
+      const isOverflowing = breadcrumbMaxWidth < totalBreadcrumbWidth
+      if (!isOverflowing || visibleItems.value.length <= 3) {
+        return
+      }
+      // Remove from the left side
+      const removed = visibleItems.value.splice(offsetIndex, 1)
+
+      hiddenItems.value.push(removed[0])
+      reduceBreadcrumb(offsetIndex)
+    }
+
+    const lastHiddenItem = computed(() =>
+      hiddenItems.value.length >= 1 ? unref(hiddenItems)[unref(hiddenItems).length - 1] : { to: {} }
+    )
+
+    const renderBreadcrumb = () => {
+      displayItems.value = props.items.slice()
```

Comment:
```suggestion
      displayItems.value = [...props.items]
```

If I don't miss anything both create a shallow copy of `props.items`. I find `[...props.items]` more intuitive.

Reply:
Agree, slice seems to minimal
faster but we usually use the spread operator for copys